src/include/cephfs/ceph_ll_client.h

```diff
@@ -71,6 +71,7 @@ struct ceph_statx {
 	struct timespec	stx_mtime;
 	struct timespec	stx_btime;
 	uint64_t	stx_version;
+	uint64_t 	stx_attributes;
```

Comment:
This is changing the ABI. Although the field has been added to the end of the structure and since this structure is used to read information, this change is probably fine, but I still wanted to flag this.

Reply:
@vshankar I suspect this ABI change could lead to unexpected behaviors: if a user of libcephfs (e.g., Samba's `vfs_ceph_new`) is compiled against the old version, but then linked at run-time with the new version, calling APIs like `ceph_ll_getattr` may cause memory corruption on the caller side (libcephfs assigns extra 8-bytes into provided `stx`). 

As a side note: the kernel's `struct statx` has few [reserved](https://elixir.bootlin.com/linux/v6.14.5/source/tools/include/uapi/linux/stat.h#L137) fields for future extensions. Maybe libcephfs should do the same?

Reply:
Very good points by all.

This was added to support querying if inode was encrypted or not. While we like to stay feature parity with original fscrypt spec, some cases (like this) make sense not to. It seems this capability will cause more harm than good.

Pushed a commit removing this addition/change.